src/shared/dissect-image.c

```diff
@@ -2262,6 +2271,11 @@ int dissected_image_acquire_metadata(DissectedImage *m) {
 
         assert(m);
 
+        /* As per the os-release spec, if the image is an extension it will have a file
+         * named after the image name in extension-release.d/ */
+        if (m->image_name)
+                paths[META_EXTENSION_RELEASE] = strjoina("/usr/lib/extension-release.d/extension-release.", m->image_name);
+
         for (; n_meta_initialized < _META_MAX; n_meta_initialized ++)
```

Comment:
this loop needs updating now that one entry of paths[] may be NULL

Reply:
and similar that other lop below. maybe just insert a `if (paths[n_meta_initialized]) continue;` at the beginning of each loop

Reply:
I can, but it actually works out just fine as-is though: the NULLSTR_FOREACH just skips it, so ```fd``` is never set and there's just a log_debug message then it skips over it:

https://github.com/systemd/systemd/blob/356b7a58f3541b014b1385757ab179b50d072603/src/shared/dissect-image.c#L2276

Wouldn't this be enough already, or am I missing something?

Reply:
it would generate a pretty confusing debug message. I'd really skip the while entry if things are NULL

Reply:
Ok, updated, PTAL